### ISSUE TITLE
Color legend doesn't appear on svg renderer

### ISSUE THREAD

**Issue by satyasinha** [vega/vega-lite#2541]
The color legend doesn't appear for quantitative measurement types on SVG renderer. This works fine on Canvas renderer but would be nice if it could work on SVG too.

To reproduce this:
1. go to https://vega.github.io/editor/#/examples/vega-lite/scatter
2. add `"color": {"field": "Miles_per_Gallon","type": "quantitative"}` to the `encoding` object
3. toggle renderers in the bottom right

**Comment by kanitw** [comment]
@arvind @jheer This looks more like a Vega bug?

**Comment by jheer** [comment]
I see the same issue in the editor, however I am unable to replicate locally using Vega v3.0.0-beta.37. Both an initial SVG render and hot-swapping the renderer from an initial Canvas view to SVG work for me. So either we need to update the editor to a new version or the error may stem from another source...

**Comment by jheer** [comment]
I think I found the problem. If you look at the generated SVG for the legend gradient it looks like this:
```
<path d="M0,0h100v20h-100Z" style="fill: url('https://vega.github.io/editor/#/edited#gradient_6'); stroke: rgb(221, 221, 221); stroke-width: 0; opacity: 1;"></path>
```
Notice the double hash? This is an invalid URL and hence the gradient lookup is failing. The SVGRenderer is creating the URL using `window.location.href` and clearly that is breaking here. Will investigate fixes...

**Comment by jheer** [comment]
Fixed by vega/vega-scenegraph@50d9da3fcdc5e3d74b32459320b630f594e2c79e and released in Vega v3.0.0-beta.38. This fix will eventually propagate to the editor.

**Comment by satyasinha** [comment]
thanks guys 👍
